### PR TITLE
RTN15e

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

**Comment by ricardopereira** [comment]
I think this should be rebased with #231 fixes.

**Comment by ricardopereira** [comment]
I think this should be rebased with #228 fixes.

File: ablySpec/RealtimeClientConnection.swift [hunk 1]
```diff
@@ -1302,6 +1302,44 @@ class RealtimeClientConnection: QuickSpec {
...
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+                        let channel = client.channels.get("test")
+
+                        channel.attach()
+                        expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+
+                        client.onDisconnected()
```
Comment (mattheworiordan) [inline ablySpec/RealtimeClientConnection.swift]: Why don't you add a `once(.Connecting)` step that explicitly resets the key to another value. That way you can be sure the iOS code is actually updating the `Connection#key`?

**Comment by ricardopereira** [comment]
Rebased with master with changes: cc419e7.

File: ablySpec/RealtimeClientConnection.swift (hunk 1 above)
Comment (ricardopereira) [inline ablySpec/RealtimeClientConnection.swift]: Done cc419e7.

**Comment by mattheworiordan** [comment]
LGTM

**Comment by tcard** [comment]
LGTM

**Pull request merged**